src/compiler/core/src/commonMain/kotlin/community/flock/wirespec/compiler/core/emit/KotlinEmitter.kt

```diff
@@ -55,240 +57,195 @@ class KotlinEmitter(
-                "${SPACER}data class Response${it.status}${
-                    it.content?.emitContentType()?.sanitizeSymbol() ?: "Unit"
-                } (override val headers: Map<String, List<Any?>>${it.content?.let { ", val body: ${it.reference.emit()}" } ?: ""} ): Response${it.status}<${it.content?.reference?.emit() ?: "Unit"}> { override val status = ${it.status}; override val content = ${it.content?.let { "Wirespec.Content(\"${it.type}\", body)" } ?: "null"}}"
-            }
-        }
-        |${
-            responses.filter { !it.status.isInt() }.distinctBy { it.status to it.content?.type }.joinToString("\n") {
-                "${SPACER}data class Response${it.status.firstToUpper()}${
-                    it.content?.emitContentType()?.sanitizeSymbol() ?: "Unit"
-                } (override val status: Int, override val headers: Map<String, List<Any?>>${it.content?.let { ", val body: ${it.reference.emit()}" } ?: ""} ): Response${it.status.firstToUpper()}<${it.content?.reference?.emit() ?: "Unit"}> { override val content = ${it.content?.let { "Wirespec.Content(\"${it.type}\", body)" } ?: "null"}}"
+
+    override fun EndpointClass.Path.emit(): String =
+        value.joinToString("/", "/") {
+            when (it) {
+                is EndpointClass.Path.Literal -> it.value
+                is EndpointClass.Path.Parameter -> "${'$'}{${it.value}}"
             }
         }
-        |${SPACER}suspend fun ${name.sanitizeSymbol().firstToLower()}(request: Request<*>): Response<*>
-        |${SPACER}companion object{
-        |${SPACER}${SPACER}const val PATH = "${path.emitSegment()}"
-        |${SPACER}${SPACER}const val METHOD = "${method.name}"
-        |${requests.emitRequestMapper()}
-        |${responses.emitResponseMapper()}
-        |${SPACER}}
-        |}
-        |""".trimMargin()
-    }
 
-    override fun Node.emitName(): String = when (this) {
-        is Endpoint -> "${this.name}Endpoint"
-        is Enum -> this.name
-        is Refined -> this.name
-        is Type -> this.name
-    }
+    override fun EndpointClass.Content.emit(): String = """
+        |Wirespec.Content("$type", body)
+    """.trimMargin()
 
-    private fun Endpoint.emitRequestSignature(content: Endpoint.Content? = null): String {
-        val pathField = path
-            .filterIsInstance<Endpoint.Segment.Param>()
-            .map { Type.Shape.Field(it.identifier, it.reference, false) }
-        val parameters = pathField + query + headers + cookies
-        return """
-            |(${
-            parameters
-                .plus(content?.reference?.toField("body", false))
-                .filterNotNull()
-                .joinToString(", ") { it.emit() }
-        })
-        """.trimMargin()
-    }
+    override fun EndpointClass.RequestMapper.emit(): String = """
+        |fun <B> $name(contentMapper: Wirespec.ContentMapper<B>) = { request: Wirespec.Request<B> ->
```

Comment:
Guess this needs a.space :P

Reply:
In front of the `|` it is not in line with the next line

Reply:
Check